libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ModelOverview.tsx

```diff
@@ -168,30 +169,36 @@ export class ModelOverview extends React.Component<
         return errorCohort.cohort.getCohortID();
       }
     );
-    if (!this.ifCohortIndexesEquals(newDatasetCohortIDs, oldDatasetCohortIDs)) {
-      const addCohortIDs = newDatasetCohortIDs.filter(
-        (x) => !oldDatasetCohortIDs.includes(x)
-      );
-      const deleteCohortIDs = oldDatasetCohortIDs.filter(
-        (x) => !newDatasetCohortIDs.includes(x)
-      );
-      if (addCohortIDs.length > 0) {
-        this.setState(
-          {
-            selectedDatasetCohorts:
-              this.state.selectedDatasetCohorts?.concat(addCohortIDs)
-          },
-          () => this.updateDatasetCohortStats()
+    if (!this.state.isDatasetCohortMetricCalcaulationInProgress) {
```

Comment:
So while this is in progress we don't allow changes to dataset cohorts if I read this right? What if the user creates or deletes a new cohort in the meanwhile? Or are we blocking the entire UI for the duration of the calculation?